packages/story-editor/src/app/highlights/quickActions/useQuickActions.js

```diff
@@ -157,18 +171,43 @@ const useQuickActions = () => {
         actionLabel: __('Undo', 'web-stories'),
         dismissable: false,
         message: __('Element properties have been reset', 'web-stories'),
-        onAction: undo,
+        // Don't pass a stale version of `undo`
+        onAction: () => undoRef.current(),
       });
     },
-    [handleResetProperties, showSnackbar, undo]
+    [handleResetProperties, showSnackbar]
   );
 
+  /**
+   * Highlights a panel in the editor. Triggers a tracking event
+   * using the selected element's type.
+   *
+   * The selected element and selected element type may be overridden
+   * using the `elementParams` arguments.
+   *
+   * @param {string} highlight The panel to highlight
+   * @param {Object} elementParams
+   * @param {string} elementParams.elementId The element id that is or will be selected in the canvas.
+   * @param {string} elementParams.elementType The type of the element that is or will be selected in the canvas.
+   * @param {Event} ev The triggering event.
+   */
   const handleFocusPanel = useCallback(
-    (highlight) => (elementId) => (ev) => {
-      ev.preventDefault();
-      setHighlights({ elementId, highlight });
-    },
-    [setHighlights]
+    (highlight) =>
+      ({ elementId, elementType } = {}) =>
+      (ev) => {
+        ev.preventDefault();
+        setHighlights({
+          elementId: elementId || selectedElement?.id,
+          highlight,
+        });
+
+        trackEvent('quick_action', {
+          name: 'focus_panel',
+          element: elementType || selectedElement?.type,
+          panel: highlight,
+        });
```

Comment:
Most quick actions focus a panel in the library or the inspector.The `selectedElement` usually does not change when clicking these actions.

The one case where it does is when there is no selected element (default state). Clicking the quick actions should select the background element and still highlight the relevant panel.

---

There's one place where we need to specify that the background _should_ be selected if it already isn't, so we pass in an override when we need to specify which element to highlight. This makes sure that the inspector has the correct panel to highlight.